### PR TITLE
Dima_feature_posts

### PR DESCRIPTION
**Pull request by velychkodmytro**
Post functionality

### CODE REVIEW

File: src/database/entities/user-entity.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  PrimaryGeneratedColumn,
+} from 'typeorm';
+import * as bcrypt from 'bcrypt';
+import { v4 as uuidv4 } from 'uuid';
+
+@Unique(['id'])
+@Unique(['email'])
+@Entity({ name: 'users' }) // Name of your table in database
+export class User {
+  @PrimaryGeneratedColumn() //Mysql doesnt have uuid
```
Comment (Mainsed) [inline src/database/entities/user-entity.ts]: I think need to delete this comment and the commented lines below

File: src/modules/user/user.service.ts [hunk 2]
```diff
@@ -1,27 +1,43 @@
...
+import { UpdateUserDto } from './dto/update-user-dto';
+import { Injectable, Inject } from '@nestjs/common';
+import { DeleteResult, Repository, UpdateResult } from 'typeorm';
+import { InjectRepository } from '@nestjs/typeorm';
+import { User } from 'src/database/entities/user-entity';
+import { USER_REPOSITORY } from 'src/database/database.constants';
 @Injectable()
 export class UserService {
-  constructor(@InjectModel(User.name) private userModel: Model<UserDocument>) {}
+  //
```
Comment (Mainsed) [inline src/modules/user/user.service.ts]: this one too

File: src/modules/user/dto/create-user-dto.ts [hunk 3]
```diff
@@ -1,4 +1,10 @@
-import { IsEmail, IsNotEmpty, Length } from 'class-validator';
+import {
+  IsAlpha,
```
Comment (Mainsed) [inline src/modules/user/dto/create-user-dto.ts]: unused isAplha and isNumber

File: src/modules/post/dto/update-post-dto.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { IsEmail, IsNotEmpty, Length, IsOptional } from 'class-validator';
+
+export class UpdatePostDto {
+  @IsOptional()
+  @IsNotEmpty()
+  @Length(2, 255)
+  title: string;
+
+  @IsOptional()
+  @IsEmail()
```
Comment (Mainsed) [inline src/modules/post/dto/update-post-dto.ts]: text isEmail is it correct?

File: src/modules/post/dto/create-post-dto.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+export class CreatePostDto {
+  @IsNotEmpty()
+  @Length(2, 255)
+  title: string;
+
+  @IsNotEmpty()
+  @Length(2, 255)
+  text: string;
+
+  // @IsNotEmpty()
```
Comment (Mainsed) [inline src/modules/post/dto/create-post-dto.ts]: commented line

File: src/database/entities/user-entity.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+  //  @UpdateDateColumn({ name: 'updated_at', type: 'timestamp', update: false })
+  //  updatedAt: Date;
+
+  //  @CreateDateColumn({ name: 'created_at', type: 'timestamp', update: false })
+  //  createdAt: Date;
+
+  @BeforeInsert()
+  async beforeInsertActions() {
+    this.id = uuidv4();
```
Comment (Mainsed) [inline src/database/entities/user-entity.ts]: I think it would be better to do similar as in my part with uuid in PrimaryGeneratedColumn

**Review by Mainsed — CHANGES_REQUESTED** [review]

File: src/database/entities/user-entity.ts (hunk 6 above)
Comment (velychkodmytro) [reply src/database/entities/user-entity.ts]: Done

File: src/modules/post/dto/create-post-dto.ts (hunk 5 above)
Comment (velychkodmytro) [reply src/modules/post/dto/create-post-dto.ts]: Done

File: src/modules/post/dto/update-post-dto.ts (hunk 4 above)
Comment (velychkodmytro) [reply src/modules/post/dto/update-post-dto.ts]: Done

File: src/modules/user/dto/create-user-dto.ts (hunk 3 above)
Comment (velychkodmytro) [reply src/modules/user/dto/create-user-dto.ts]: Done

File: src/modules/user/user.service.ts (hunk 2 above)
Comment (velychkodmytro) [reply src/modules/user/user.service.ts]: Done

File: src/database/entities/user-entity.ts (hunk 1 above)
Comment (velychkodmytro) [reply src/database/entities/user-entity.ts]: Done

**Pull request closed without merging**